### ISSUE TITLE
Unrecognized targets in BSP server mode

### ISSUE THREAD

**Issue by odisseus** [propensive/fury#1077]
## Steps to reproduce
1. Launch Fury as a BSP server, e. g. using IntelliJ as a client.
2. Request compilation of one of the intermediate modules e. g. `fury/model`.
3. Request compilation of `fury/frontend`

**Expected behaviour:** Compilation is completed without errors both times
**Actual behaviour:** Lots of stack traces are printed during the second compilation, but it still succeeds

It seems that exceptions are thrown for every dependency of `fury/frontend` which is not a dependency of `fury/model`.

## Example stack trace
```
Feb 17, 2020 7:58:48 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleNotification
WARNING: Notification threw an exception: {
"jsonrpc": "2.0",
"method": "build/taskStart",
"params": {
"taskId": {
"id": "23"
},
"eventTime": 1581965928620,
"message": "Compiling default_eucalyptus_core (1 Scala source)",
"dataKind": "compile-task",
"data": {
"target": {
"uri": "file:/home/virtuslab/devel/fury/.fury/work/default_eucalyptus_core/?id\u003ddefault_eucalyptus_core"
}
}
}
}
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:152)
at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:220)
at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:187)
at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:192)
at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:99)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
at sun.reflect.GeneratedMethodAccessor2.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
... 11 more
Caused by: java.util.NoSuchElementException: key not found: TargetId(default_eucalyptus_core)
at scala.collection.MapLike.default(MapLike.scala:235)
at scala.collection.MapLike.default$(MapLike.scala:234)
at scala.collection.AbstractMap.default(Map.scala:63)
at scala.collection.MapLike.apply(MapLike.scala:144)
at scala.collection.MapLike.apply$(MapLike.scala:143)
at scala.collection.AbstractMap.apply(Map.scala:63)
at fury.core.FuryBuildClient.onBuildTaskStart(compile.scala:384)
... 15 more
```

**Comment by propensive** [comment]
Are the stack traces visible to users?

**Comment by odisseus** [comment]
Yes

**Comment by odisseus** [comment]
The problem lies in the fact that the build client stores a `Compilation` object and uses its data to process messages from the build server. Since #949, the connection may be reused for a different compilation task, which may have very little in common with the previous compilation.

**Comment by odisseus** [comment]
Fixed as of commit e556afa6adc8f428b8a83bc0bfabdaf8d6071a8e. However I had to edit the BSP discovery file manually to update the path to the Fury executable.
